docs/fuse-box/tutorials/api-trade-quotes.mdx

```diff
@@ -0,0 +1,338 @@
+
+async function fetchData() {
+    try {
+      const response = await fetch(apiUrl);
+      if (!response.ok) {
+        throw new Error('Network response was not ok');
+      }
+      const data = await response.json();
+      console.log(data);
+      } catch (error) {
+      console.error('There was a problem with the fetch operation:', error);
+    }
+  }
+
+fetchData();
+```
+
+Save the file.
+
+Run the application using the command:
+
+```bash
+npm run dev
+```
+Open your browser, go to `http://localhost:3000`, and open the developer console. You will find the results logged to the console.
+
+## Step 4: Implementing the Application
+### State Management with React Hooks
+We use the `useState` hook to manage the form inputs and API response data. This allows us to keep track of the user inputs and the API results.
+
+```javascript
+const [inputs, setInputs] = useState({
+  sellToken: 'FUSE',
+  buyToken: '',
+  sellAmount: '1000000000000000000'
+});
+const [result, setResult] = useState(null);
+```
+
+- `inputs` stores the values of the form fields.
+- `result` will hold the data fetched from the API.
+
+### Handling Form Input Changes
+The `handleInputChange` function updates the `input` state whenever a user types into the form fields. This function is triggered on every keystroke, ensuring that the state is always up-to-date with what the user has entered.
+
+```javascript
+const handleInputChange = (event) => {
+  const { name, value } = event.target;
+  setInputs(prevState => ({
+    ...prevState,
+    [name]: value
+  }));
+};
+```
+
+We use a function to ensure the new state is based on the correct previous state, which is important in React to prevent state update issues during rapid state changes.
+
+### Submitting the Form
+The `handleSubmit` function is called when the user submits the form. It prevents the default form submission event, constructs the API request, and handles the API response.
+
+```javascript
+const handleSubmit = async (event) => {
+  event.preventDefault();
+  const { sellToken, buyToken, sellAmount } = inputs;
+  const url = `https://api.fuse.io/api/v1/trade/quote`;
+  const params = { apiKey: 'your_api_key', sellToken, buyToken, sellAmount };
+
+  try {
+    const response = await axios.get(url, { params });
+    setResult(response.data);
+  } catch (error) {
+    console.error('Error fetching trade quote:', error);
+    setResult({ error: error.message });
+  }
+};
+```
+
+We make an asynchronous API call using Axios. Upon success, the result is stored in the `result` state, and if an error occurs, it is handled.
+
+## Step 5: Building the User Interface
+The user interface uses HTML form elements styled with Tailwind CSS. It includes input fields for the **sellToken**, **buyToken**, and **sellAmount**, and a submit button.
+
+```javascript
+<form onSubmit={handleSubmit} className="space-y-4 bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4">
```

Comment:
Where i should put this form? To return? To separate component?

Reply:
Putted it into return and got many errors.
I full code found that you used for additional div to wrap this form, please fix

Reply:
Fixed.